pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -611,7 +611,9 @@ func (gc *GarbageCollector) deleteObject(item objectReference) error {
 	}
 	uid := item.UID
 	preconditions := v1.Preconditions{UID: &uid}
-	deleteOptions := v1.DeleteOptions{Preconditions: &preconditions}
+	// when GC deletes an object, it's always a cascading deletion.
+	falseVar := false
+	deleteOptions := v1.DeleteOptions{Preconditions: &preconditions, OrphanDependents: &falseVar}
```

Comment:
There is no default orphan policy.

I think we should only either orhpan dependents, or recursively cascading deletion to the deepest level.